src/ctapipe/io/hdf5monitoringsource.py

```diff
@@ -0,0 +1,577 @@
+    ...     print(event.monitoring.tel[tel_id].camera.coefficients["is_valid"])
+    40587.000000011576
+    [[0.01539444 0.01501589 0.0158232  ... 0.01514254 0.01504862 0.01497081]
+     [0.25207437 0.24654945 0.25933876 ... 0.24859268 0.24722679 0.24587582]]
+    [[399.5        398.66666667 399.5        ... 399.25       398.41666667
+      399.        ]
+     [400.08333333 400.41666667 399.91666667 ... 400.25       399.5
+      399.66666667]]
+    [[ 0.01000023  0.1800003  -0.09000015 ... -0.12999916  0.1800003
+       0.07999992]
+     [ 0.2800007  -0.27000046  0.11000061 ...  0.04000092 -0.19000053
+      -0.4699993 ]]
+    [[False False False ... False False False]
+     [False False False ... False False False]]
+    True
+
+    Attributes
+    ----------
+    input_files: list of Paths
+        Paths to the input monitoring files.
+    pixel_statistics: dict
+        Dictionary to hold pixel statistics tables
+    camera_coefficients: dict
+        Dictionary to hold camera coefficients
+    telescope_pointings: dict
+        Dictionary to hold telescope pointing information
+
+    """
+
+    input_files = List(
+        Path(exists=True, directory_ok=False),
+        default_value=[],
+        help="List of paths to the HDF5 input files containing monitoring data",
+    ).tag(config=True)
+
+    def __init__(
+        self, subarray=None, input_files=None, config=None, parent=None, **kwargs
+    ):
+        """
+        MonitoringSource for monitoring files in the standard HDF5 data format
+
+        Parameters:
+        -----------
+        subarray : SubarrayDescription or None
+            Description of the subarray to use for compatibility checks. If None,
+            the subarray description is read from the input files and checked for consistency.
+        input_files : Path, or list of Paths, optional
+            Path(s) of the files to load. Can be a single Path, or a list of Paths.
+            These will be added to any files already configured via the configuration system.
+        config : traitlets.loader.Config
+            Configuration specified by config file or cmdline arguments.
+            Used to set traitlet values.
+            Set to None if no configuration to pass.
+        parent:
+            Parent from which the config is used. Mutually exclusive with config
+        kwargs
+        """
+
+        super().__init__(
+            subarray=subarray,
+            config=config,
+            parent=parent,
+            **kwargs,
+        )
+        from ..monitoring import (
+            FlatfieldImageInterpolator,
+            FlatfieldPeakTimeInterpolator,
+            PedestalImageInterpolator,
+            PointingInterpolator,
+        )
+
+        # Set the subarray description
+        self.subarray = subarray
+
+        # Handle additional input_files parameter - extend the existing list
+        if input_files is not None:
```

Comment:
I think this is highly surprising how you implement this behavior here.

Instead of passing the traitlet kwarg to super, you treat it specially by extending.

This is inconsistent with all other ctapipe class, where passing a tratilet to `__init__` overrides it (it is the way of setting the configurable thing with the highest precedence).

Please remove `input_files` from the `__init__` and just let it be handled by the component system.